### DISCUSSION TITLE
What IDE/Plugin we should use to write/update scripts

### DISCUSSION THREAD

**Discussion by dhanan-np** [JanssenProject/jans#4721] [Q&A] [answered]
Can somebody please help with, what JAVA IDE/plugin we should use for creating/editing scripts. So that is becomes easy and error-free for us to make any changes as per our requirement. 
Would be great if anyone can help me with any reference where it is discussed in detailed

**Comment by ossdhaval** [comment]
IDE choice is a developer preference and is sometimes dictated by the programing language that you are using. IDEs like IntelliJ and Eclipse are pretty robust and can support Java as well as Jython.

**Comment by nynymike** [comment, marked as answer]
Remember, that when you are debugging scripts, the container injects a lot of contextual information. It is hard to mock all this information in an IDE. We recommend connecting your IDE to a remote servlet container and using breaks to debug at runtime. Did you review [these docs](https://docs.jans.io/v1.0.12/admin/developer/interception-scripts-debug/#install-an-ide) ?

**Reply by dhanan-np** [reply]
Hey Mike, thanks...
That is what exactly I wanted to know. Because it is not only about the portion of the script but also the dependencies and context.
